compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -4732,30 +4732,19 @@ static TR::Register * inlineSinglePrecisionSQRT(TR::Node *node, TR::CodeGenerato
   return node->getRegister();
 }
 
-TR::Register* OMR::X86::TreeEvaluator::performSimpleAtomicMemoryUpdate(TR::Node* node, int8_t size, TR_X86OpCodes op, TR::CodeGenerator* cg)
+TR::Register* OMR::X86::TreeEvaluator::performSimpleAtomicMemoryUpdate(TR::Node* node, TR_X86OpCodes op, TR::CodeGenerator* cg)
    {
-   // There are two versions of native exchange helpers:
-   //     2 nodes: exchange with [address]
-   //     3 nodes: exchange with [address+offset]
-   TR::Register* object = cg->evaluate(node->getFirstChild());
-   TR::Register* offset = node->getNumChildren() == 2 ? NULL : cg->evaluate(node->getSecondChild());
-   TR::Register* value = TR::TreeEvaluator::intOrLongClobberEvaluate(node->getLastChild(), size == 8, cg);
-   // Assume that the offset is positive and not pathologically large (i.e., > 2^31).
-   if (offset && TR::Compiler->target.is32Bit() && size == 8)
-      {
-      offset = offset->getLowOrder();
-      }
+   TR::Register* address = cg->evaluate(node->getChild(0));
+   TR::Register* value   = cg->evaluate(node->getChild(1));
+   TR::Register* result  = cg->allocateRegister();
 
-   generateMemRegInstruction(op, node, generateX86MemoryReference(object, offset, 0, cg), value, cg);
+   generateRegRegInstruction(MOVRegReg(), node, result, value, cg);
```

Comment:
The original implementation used a clobberEvaluate but this version will always generate a register copy regardless of node count.  Is that because you no longer have the size of the data to determine which clobberEvaluate to call?  If passing in the size means eliminating an instruction in some cases then I think that's an acceptable tradeoff.